File: src/vs2012/SolidEdge.EmptyAddInVB/MyAddIn.vb
Imports SolidEdgeContrib.AddIn
Imports SolidEdgeContrib.Extensions
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Runtime.InteropServices
Imports System.Text
Imports System.Windows.Forms

' Please see https://github.com/SolidEdgeCommunity/SolidEdgeContrib/wiki for documentation.
Namespace $safeprojectname$

    <ComVisible(True), Guid("$guid3$"), ProgId("$safeprojectname$.MyAddin")> _
    Public Class MyAddIn
        Inherits SolidEdgeContrib.AddIn.SolidEdgeAddIn

        Public Overrides Sub OnConnection(ByVal application As SolidEdgeFramework.Application, ByVal ConnectMode As SolidEdgeFramework.SeConnectMode, ByVal AddInInstance As SolidEdgeFramework.AddIn)
            MyBase.OnConnection(application, ConnectMode, AddInInstance)

            ' Put your custom OnConnection code here.
        End Sub

        Public Overrides Sub OnConnectToEnvironment(ByVal environment As SolidEdgeFramework.Environment, ByVal firstTime As Boolean)
            MyBase.OnConnectToEnvironment(environment, firstTime)

            ' Put your custom OnConnectToEnvironment code here.
        End Sub

        Public Overrides Sub OnDisconnection(ByVal DisconnectMode As SolidEdgeFramework.SeDisconnectMode)
            MyBase.OnDisconnection(DisconnectMode)

            ' Put your custom OnDisconnection code here.
        End Sub

        Public Overrides Sub OnInitializeRibbon(ByVal ribbon As Ribbon, ByVal firstTime As Boolean)
            ' Let the base class handle initializing the ribbon via addin.manifest.
            MyBase.OnInitializeRibbon(ribbon, firstTime)

            ' Put your custom OnInitializeRibbon code here.
        End Sub

        Public Overrides Sub OnRibbonControl(ByVal ribbonControl As RibbonControl)
            ' Base class doesn't do anything special when a ribbon control is invoked unless a macro is assigned.
            MyBase.OnRibbonControl(ribbonControl)

            ' Put your custom OnInitializeRibbon code here.
        End Sub

        Public Overrides ReadOnly Property NativeResourcesDllPath() As String
            Get
                ' You can override the path to your native images if you need.
                Return MyBase.NativeResourcesDllPath
            End Get
        End Property

        <ComRegisterFunction> _
        Public Shared Sub OnRegister(ByVal t As Type)
            Try
                ' SolidEdgeAddIn.Register() will throw an exception if it cannot locate an embedded resource named [DEFAULT_NAMESPACE].addin.manifest.
                ' If you want to take control of the registration process, simply don't call SolidEdgeAddIn.Register().
                SolidEdgeContrib.AddIn.SolidEdgeAddIn.Register(t)
            Catch ex As System.Exception
                MessageBox.Show(ex.StackTrace, ex.Message)
            End Try

            ' Perform any additional registration procedures if needed.
        End Sub

        <ComUnregisterFunction> _
        Public Shared Sub OnUnregister(ByVal t As Type)
            SolidEdgeContrib.AddIn.SolidEdgeAddIn.Unregister(t)
        End Sub

    End Class

End Namespace

File: src/vs2012/SolidEdge.EmptyAddIn/MyAddIn.cs
using SolidEdgeContrib.AddIn;
using SolidEdgeContrib.Extensions;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.InteropServices;
using System.Text;
using System.Windows.Forms;

// Please see https://github.com/SolidEdgeCommunity/SolidEdgeContrib/wiki for documentation.
namespace $safeprojectname$
{
    [ComVisible(true)]
    [Guid("$guid3$")]
    [ProgId("$safeprojectname$.MyAddin")]
	public class MyAddIn : SolidEdgeContrib.AddIn.SolidEdgeAddIn
	{
        public override void OnConnection(SolidEdgeFramework.Application application, SolidEdgeFramework.SeConnectMode ConnectMode, SolidEdgeFramework.AddIn AddInInstance)
        {
            base.OnConnection(application, ConnectMode, AddInInstance);

            // Put your custom OnConnection code here.
        }

        public override void OnConnectToEnvironment(SolidEdgeFramework.Environment environment, bool firstTime)
        {
            base.OnConnectToEnvironment(environment, firstTime);

            // Put your custom OnConnectToEnvironment code here.
        }

        public override void OnDisconnection(SolidEdgeFramework.SeDisconnectMode DisconnectMode)
        {
            base.OnDisconnection(DisconnectMode);

            // Put your custom OnDisconnection code here.
        }

        public override void OnInitializeRibbon(Ribbon ribbon, bool firstTime)
        {
            // Let the base class handle initializing the ribbon via addin.manifest.
            base.OnInitializeRibbon(ribbon, firstTime);

            // Put your custom OnInitializeRibbon code here.
        }

        public override void OnRibbonControl(RibbonControl ribbonControl)
        {
            // Base class doesn't do anything special when a ribbon control is invoked unless a macro is assigned.
            base.OnRibbonControl(ribbonControl);

            // Put your custom OnRibbonControl code here.
        }

        public override string NativeResourcesDllPath
        {
            get
            {
                // You can override the path to your native images if you need.
                return base.NativeResourcesDllPath;
            }
        }

        [ComRegisterFunction]
        public static void OnRegister(Type t)
        {
            try
            {
                // SolidEdgeAddIn.Register() will throw an exception if it cannot locate an embedded resource named [DEFAULT_NAMESPACE].addin.manifest.
                // If you want to take control of the registration process, simply don't call SolidEdgeAddIn.Register().
                SolidEdgeAddIn.Register(t);
            }
            catch (System.Exception ex)
            {
                MessageBox.Show(ex.StackTrace, ex.Message);
            }

            // Perform any additional registration procedures if needed.
        }

        [ComUnregisterFunction]
        public static void OnUnregister(Type t)
        {
            SolidEdgeAddIn.Unregister(t);
        }
	}
}
